### PR TITLE
chore(workflows): configure tests workflow

### PR DESCRIPTION
**Pull request by jgolden17**
This adds a workflow to run test cases again pull requests to the main branch

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '10.x', '12.x' ]
+    name: Node ${{ matrix.node }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        git remote set-branches --add origin main
```
Comment (igetgames) [inline .github/workflows/tests.yml:17]: I've seen other `one` projects do this, but I have not seen this on the checkout action for main branches. Does an error occur if it's missing?

Comment (jgolden17) [reply .github/workflows/tests.yml:17]: I haven't tested without it as I stole this directly from @nellyk. @nellyk, is `actions/checkout@v2` doing a shallow clone here?

Comment (igetgames) [reply .github/workflows/tests.yml:17]: If there was an issue with the default branch not being `master` before, I believe it was resolved in actions/checkout#305 and the above is no longer necessary.

Re: shallow clones `actions/checkout` will only grab a single commit unless you provide `fetch-depth: 0` as a parameter.

**Comment by nellyk** [comment]
@jgolden17  please merge this before deleting travis or you can delete Travis on this PR too. I have enabled github actions for testing

**Comment by marcusrbrown** [comment]
Doop.

**Comment by marcusrbrown** [comment]
@jgolden17 Was one of your registered email addresses removed? I guess the CLA Bot decided that meant you are not you. Well, are you?

Probably easiest reopen (if you need to re-sign the CLA anyway).

**Comment by nellyk** [comment]
closing this as it is superseeded by https://github.com/americanexpress/react-seo/pull/50

**Pull request closed without merging**